store/governance_proposals.store.ts

```diff
@@ -281,15 +331,25 @@ export const useGovernanceProposalsStore = defineStore({
         proposal.createdTimestamp = Number(block.timestamp);
         proposal.createdDatetimeFormatted = formatDate(new Date(Number(block.timestamp) * 1000));
         proposal.createdBlockNumber = event.blockNumber;
-
-        // const voteStartDate = new Date(Number(proposal.voteStart) * 1000);
-        // const voteEndDate = new Date(Number(proposal.voteEnd) * 1000);
-        // console.log("Vote Start Date: ", voteStartDate);
-        // console.log("Vote End Date: ", voteEndDate);
+        // keep track of the proposal executed timestamp and block number if the proposal is executed
+        const executedProposal = this.fundProposals[this.web3Store.chainId][this.fundStore.fund?.address][proposal.proposalId];
+        proposal.executedTimestamp = executedProposal?.executedTimestamp;
+        proposal.executedBlockNumber = executedProposal?.executedBlockNumber;
 
         const proposalState = await this.fundStore.fundGovernorContract.methods.state(proposal.proposalId).call();
         proposal.state = ProposalStateMapping[proposalState]
 
+        console.log("proposal: ", proposal);
+
+        // If the clock mode is block number, we have to check a timestamp for the block number.
+        if(this.fundStore.fund?.clockMode?.mode === ClockMode.BlockNumber) {
+          const blockStart = await this.fundStore.web3.eth.getBlock(proposal.voteStart);
+          const blockEnd = await this.fundStore.web3.eth.getBlock(proposal.voteEnd);
```

Comment:
voteEnd can be in the future also and this will crash if there is no blockEnd
you can test it by creating a new proposal

if there is no blockEnd I guess you have to estimate the voteEnd timestamp by the current blocks per minute number
maybe try to figure how they do it in tally? @cinquemb do you have any idea?

Reply:
One can use the proposal duration info added to the start block to estimate time end. Having a config file for storing blocks per second for each chain could be useful for this too.

Reply:
> One can use the proposal duration info added to the start block to estimate time end. Having a config file for storing blocks per second for each chain could be useful for this too.
if you meant the `votingPeriod` the problem is that it is also expressed in the blocknumbers if clock mode is blocknumber.
![image](https://github.com/user-attachments/assets/3747faf9-26cd-4be3-b582-0f6d2e5684a0)

https://docs.openzeppelin.com/contracts/4.x/api/governance#IGovernor-votingPeriod--

Reply:
Yes, in that case, you will still need the blocks per second for the chain